src/chi-vue/src/components/checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,62 @@
+import { Component, Vue, Prop } from 'vue-property-decorator';
+import { CHECKBOX_CLASSES, SR_ONLY } from '@/constants/classes';
+import { GENERIC_EVENTS } from '@/constants/events';
+import { CheckboxState } from '@/constants/types';
+
+@Component({})
+export default class Checkbox extends Vue {
+  @Prop({ required: true }) id!: string;
+  @Prop() label?: string;
+  @Prop() name?: string;
+  @Prop() selected?: CheckboxState;
+  @Prop() disabled?: boolean;
+
+  state = this.selected;
+
+  _emitChange(ev: Event) {
+    this.$emit(GENERIC_EVENTS.CHANGE, ev);
+  }
+
+  _updateCheckboxState() {
+    const checkbox = this.$refs.checkbox as HTMLInputElement;
+
+    if (this.state === 'indeterminate') {
+      checkbox.indeterminate = true;
+      checkbox.classList.add(CHECKBOX_CLASSES.INDETERMINATE);
+    } else {
+      checkbox.indeterminate = false;
+      checkbox.classList.remove(CHECKBOX_CLASSES.INDETERMINATE);
+    }
+  }
+
+  mounted() {
+    this._updateCheckboxState();
+  }
+
+  updated() {
+    this._updateCheckboxState();
+  }
```

Comment:
Can this be done using a watcher which checks the state?

Reply:
Seems like no. Setting the indeterminate state happens on the actual HTMLInputElement like this:
https://www.w3schools.com/jsref/prop_checkbox_indeterminate.asp